vault/resource_consul_secret_backend.go

```diff
@@ -296,13 +296,17 @@ func consulSecretsBackendCustomizeDiff(_ context.Context, diff *schema.ResourceD
 	newToken := diff.Get("token").(string)
 	newBootstrap := diff.Get("bootstrap").(bool)
 
-	// If the user sets bootstrap to false but doesn't provide a token, disallow it.
-	if newToken == "" && !newBootstrap {
+	// Disallow the following:
+	//   1. Bootstrap is false, the token field is empty, and we know this is the final value of token.
+	if !newBootstrap && newToken == "" && diff.NewValueKnown("token") {
 		return fmt.Errorf("field 'bootstrap' must be set to true when 'token' is unspecified")
 	}
 
-	// If the user sets bootstrap to true and also provides a token, disallow it.
-	if newToken != "" && newBootstrap {
+	// Disallow the following:
+	//   1. Bootstrap is true and the token field is set to something.
+	//   2. Bootstrap is true and the token field is empty, but we don't know the final value of token.
+	if newBootstrap && newToken != "" ||
+		(newBootstrap && newToken == "" && !diff.NewValueKnown("token")) {
 		return fmt.Errorf("field 'bootstrap' must be set to false when 'token' is specified")
 	}
```

Comment:
This if statement can actually be simplified to:
```go
if newBootstrap &&
		(newToken != "" || newToken == "" && !diff.NewValueKnown("token")) {
		return fmt.Errorf("field 'bootstrap' must be set to false when 'token' is specified")
	}
```

but I actually find the first version to be easier to read logic-wise. Open to changing it though.